plugins/forc-index/src/ops/forc_index_build.rs

```diff
@@ -226,3 +228,40 @@ pub fn init(command: BuildCommand) -> anyhow::Result<()> {
 
     Ok(())
 }
+
+fn verbose_error_message() -> String {
+    let mut error = "❌ Build failed.".to_string();
+
+    if cfg!(target_arch = "aarch64") {
+        let mut extra_msg = String::new();
+
+        let env_vars: HashSet<_> = env::vars().map(|(k, _)| k).collect();
+
+        if !env_vars.contains("LIBCLANG_PATH") {
+            extra_msg.push_str("\nexport LIBCLANG_PATH='/opt/homebrew/opt/llvm/lib'");
+        }
+
+        if !env_vars.contains("LDFLAGS") {
+            extra_msg.push_str("\nexport LDFLAGS='-L/opt/homebrew/opt/llvm/lib'");
+        }
+
+        if !env_vars.contains("CPPFLAGS") {
+            extra_msg.push_str("\nexport CPPFLAGS='-I/opt/homebrew/opt/llvm/include'");
+        }
+
+        if !extra_msg.is_empty() {
+            extra_msg.insert_str(
+                0,
+                r#"
+For Apple Silicon macOS users the preinstalled llvm has limited WASM targets,
+please install a binary with better support from Homebrew (brew install llvm)
+and configuring rustc with necessary environment variables:
```

Comment:
```suggestion
For Apple Silicon macOS users, the preinstalled llvm has limited WASM targets.
Please install a binary with better support from Homebrew (brew install llvm)
and configure rustc with the necessary environment variables:
```

Minor grammar corrections.